parsl/executors/high_throughput/zmq_pipes.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+
+import zmq
+import uuid
+import time
+import pickle
+
+
+class TasksOutgoing(object):
+    """ Outgoing task queue from MPIX
+    """
+    def __init__(self, task_q):
+
+        self.task_q = task_q
+        self.context = zmq.Context()
+        self.zmq_socket = self.context.socket(zmq.DEALER)
+        self.zmq_socket.bind(self.task_q)
+
+    def put(self, message):
+        # print("IN put")
+        self.zmq_socket.send_pyobj(message)
+        # print("DONE put")
+
+    def close(self):
+        self.zmq_socket.close()
+        self.context.term()
+
+
+class ResultsIncoming(object):
+
+    def __init__(self, results_q):
+        self.results_q = results_q
+
+        self.context = zmq.Context()
+        self.results_receiver = self.context.socket(zmq.DEALER)
+        self.results_receiver.bind(self.results_q)
+
+    def get(self, block=True, timeout=None):
+        result = self.results_receiver.recv_pyobj()
+        return result
+
+    def request_close(self):
+        status = self.results_receiver.send(pickle.dumps(None))
+        time.sleep(0.1)
+        return status
+
+    def close(self):
+        self.results_receiver.close()
+        self.context.term()
+
+
+class JobsQIncoming(object):
+
+    def __init__(self, task_url, server_id=None):
+
+        self.server_id = server_id if server_id else uuid.uuid4()
+        self.task_url = task_url
+
+        print("I am server #%s" % (self.server_id))
```

Comment:
log rather than print?

Reply:
This is in some dead code that I've now removed.